src/config.rs

```diff
@@ -33,6 +35,20 @@ impl Default for BuildTool {
     }
 }
 
+#[derive(Deserialize, Debug, PartialEq)]
+#[serde(tag = "type")]
+pub enum Repository {
```

Comment:
What does the toml look like for this? Something like this?

```toml
name = "lib"
repository = { type = "github", url = "https://github.com/one/two" }
```

If it's GitHub we probably don't need a `url` but a `user` and `repo`?

```toml
name = "lib"
repository = { type = "github", user = "gleam-lang", repo = "gleam" }
```
```

Reply:
Yeah, the top one is the current state. I agree your suggest looks better and we can certainly aim for that. I guess the only downside is if we want to account for enterprise deployments of GitHub. In some ways the "GitHub" here is to indicate the URL format for linking to lines rather than to say it has to be one `github.com`. That said, your one looks nicer :)